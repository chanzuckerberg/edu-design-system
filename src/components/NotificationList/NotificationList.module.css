@import '../../design-tokens/mixins.css';
/*------------------------------------*\
    #NOTIFICATION LIST
\*------------------------------------*/

/**
 * 1)  Notification list contains notification items, a count of unread notifications, and a button to mark all notifications as read
 */
.notification-list {
  @mixin reset;
  display: flex;
  flex-direction: column;
}
/**
* Individual notification list item
*/
.notification-list__item {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  padding: var(--eds-size-2);
  background-color: var(--eds-color-brand-blue-100); /* TODO */
  border: 1px solid var(--eds-color-neutral-200); /* TODO */
}

.notification-list__is-read {
  background-color: var(--eds-color-neutral-white); /* TODO */
}
/* 
* Notification list item content
* 1) The wrapper for the text
*/
.notification-list__content {
  color: var(--eds-color-neutral-600); /* TODO */
  font-size: var(--eds-font-size-xs);
}

/*
* Notification list item title
* 1) Main text title of the notification
*/
.notification-list__title a {
  color: var(--eds-color-brand-blue-700); /* TODO */
  font-size: var(--eds-font-size-sm);

  /**
  *  Notification List Item (is read)
  * 1) When isRead === true, the title link text changes to grey
  */
  .notification-list__is-read & {
    color: var(--eds-color-neutral-600); /* TODO */
  }
}

/**
* Notification list item link
*  1) Anchor that wraps the title; linked to source of notification.
*/
.notification-list__link {
  text-decoration: none;
}

/*
* Notification list item subtitle
* 1) Contains the created date (relative) and the originating source of the notification

* Notification list item button
* 1) Indicator that displays the item's read/not read status in
* a <button> element; when clicked, it toggles between read/not
* read statuses
*/
.notification-list__button {
  position: relative;
  top: var(--eds-size-1);
  appearance: none;
  border: none;
  outline: none;
  padding: 0;
  border-radius: var(--eds-theme-border-radius);
  height: 0.5rem;
  width: 0.5rem;
  background-color: var(--eds-color-brand-blue-700); /* TODO */
  cursor: pointer;

  /*
    * Notification List Item (is read)
    * 1) When isRead === true, button's background and border are changed to neutral
    */
  .notification-list__is-read & {
    background-color: var(--eds-color-neutral-white); /* TODO */
    border: 1px solid var(--eds-color-neutral-600); /* TODO */
  }
}
