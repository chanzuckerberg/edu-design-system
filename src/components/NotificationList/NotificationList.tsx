import clsx from 'clsx';
import React, { ReactNode } from 'react';
import styles from './NotificationList.module.css';
import NotificationListItem from '../NotificationListItem/NotificationListItem';

export interface Props {
  className?: string;
  children?: ReactNode;
}

export const NotificationList = ({ className, children, ...other }: Props) => {
  const componentClassName = clsx(styles['notification-list'], className, {});
  return <div className={componentClassName}>{children}</div>;
};
