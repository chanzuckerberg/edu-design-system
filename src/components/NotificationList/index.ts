export {NotificationList as default} from './NotificationList';
