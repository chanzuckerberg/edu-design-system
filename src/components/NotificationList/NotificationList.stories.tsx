import { Story, Meta } from '@storybook/react';
import React from 'react';

import { NotificationList, Props } from './NotificationList';
import NotificationListItem from '../NotificationListItem/NotificationListItem';

export default {
  title: 'Molecules/Lists/NotificationList',
  component: NotificationList,
} as Meta;

const Template: Story<Props> = (args) => (
  <NotificationList>
    <NotificationListItem
      link="#"
      sender="English Teacher"
      description="gave you feedback"
      date="now"
      source="Outsiders on Trial: Self Awareness - Trial Brief Outline"
      {...args}
    />
    <NotificationListItem
      link="#"
      sender="Math Teacher"
      description="gave you feedback"
      date="now"
      source="Unit 3: Measuring Circles - Identifying Circumference and Diameter"
      {...args}
    />
  </NotificationList>
);

export const Default = Template.bind({});
Default.args = {};

export const MarkedAsRead = Template.bind({});
MarkedAsRead.args = {
  markedAsRead: true,
};
