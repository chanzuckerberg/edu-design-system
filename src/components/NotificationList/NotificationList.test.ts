import {generateSnapshots} from '@chanzuckerberg/story-utils';
import * as stories from './NotificationList.stories';

describe('<NotificationList />', () => {
  generateSnapshots(stories);
});
