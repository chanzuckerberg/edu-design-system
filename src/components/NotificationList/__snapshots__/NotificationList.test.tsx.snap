// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationList /> Default story renders snapshot 1`] = `
<ul
  class="notification-list"
>
  <li
    class="notification-list__item"
  >
    <div
      class="notification-list__content"
    >
      <div
        class="notification-list__title"
      >
        <a
          class="notification-list__link"
          href="#"
        >
          <h6
            class="heading heading--size-h6"
          >
            English Teacher gave you feedback
          </h6>
        </a>
      </div>
      now
       路 
      Outsiders on Trial: Self Awareness - Trial Brief Outline
    </div>
    <button
      aria-label="Mark as read"
      class="notification-list__button"
    />
  </li>
  <li
    class="notification-list__item"
  >
    <div
      class="notification-list__content"
    >
      <div
        class="notification-list__title"
      >
        <a
          class="notification-list__link"
          href="#"
        >
          <h6
            class="heading heading--size-h6"
          >
            Math Teacher gave you feedback
          </h6>
        </a>
      </div>
      now
       路 
      Unit 3: Measuring Circles - Identifying Circumference and Diameter
    </div>
    <button
      aria-label="Mark as read"
      class="notification-list__button"
    />
  </li>
</ul>
`;

exports[`<NotificationList /> MarkedAsRead story renders snapshot 1`] = `
<ul
  class="notification-list"
>
  <li
    class="notification-list__item"
  >
    <div
      class="notification-list__content"
    >
      <div
        class="notification-list__title"
      >
        <a
          class="notification-list__link"
          href="#"
        >
          <h6
            class="heading heading--size-h6"
          >
            English Teacher gave you feedback
          </h6>
        </a>
      </div>
      now
       路 
      Outsiders on Trial: Self Awareness - Trial Brief Outline
    </div>
    <button
      aria-label="Mark as read"
      class="notification-list__button"
    />
  </li>
  <li
    class="notification-list__item"
  >
    <div
      class="notification-list__content"
    >
      <div
        class="notification-list__title"
      >
        <a
          class="notification-list__link"
          href="#"
        >
          <h6
            class="heading heading--size-h6"
          >
            Math Teacher gave you feedback
          </h6>
        </a>
      </div>
      now
       路 
      Unit 3: Measuring Circles - Identifying Circumference and Diameter
    </div>
    <button
      aria-label="Mark as read"
      class="notification-list__button"
    />
  </li>
</ul>
`;
